### PR TITLE
Adding support for up to 6 related posts

### PR DESCRIPTION
**Pull request by roccotripaldi**
Works with https://github.com/Automattic/wp-calypso/pull/30224 to allow up to six related posts items.

Fixes https://github.com/Automattic/jetpack/issues/11014

#### Testing instructions:
Run this on both a site that has less than 10 posts as well as a site that has more than 10 posts.

#### Proposed changelog entry for your changes:

* The Related Posts block will now show up to 6 posts.

### CODE REVIEW

**Review by aldavigdis — DISMISSED** [review]
I would say we have gone through every bit of code here and I have no objections myself.

File: modules/related-posts/jetpack-related-posts.php [hunk 1]
```diff
@@ -270,82 +357,45 @@ public function render_block( $attributes ) {
...
+			$this->render_block_row( $upper_row_posts, $block_attributes );
+			if ( $display_lower_row ) {
+				$this->render_block_row( $lower_row_posts, $block_attributes );
+			}
+			?>
+		</nav>
 		<?php
 		$html = ob_get_clean();
 
+		remove_filter( 'the_content', 'wpautop' );
```
Comment (lezama) [inline modules/related-posts/jetpack-related-posts.php]: I unsure about this change.
Could we achieve the same result by hiding `<p>`s via css.
or as @enejb suggested maybe replacing new lines from `$html` after `$html = ob_get_clean();`?

Comment (aldavigdis) [reply modules/related-posts/jetpack-related-posts.php]: I can modify this a bit — or simply remove it in anticipation for https://core.trac.wordpress.org/ticket/45495 being closed in time.

Comment (jeherve) [reply modules/related-posts/jetpack-related-posts.php]: Could you provide a bit of background on this change? Is this already an issue with the current version of the Related Posts block? If yes, how can I reproduce the problem? If not, what changed?

Comment (jeherve) [reply modules/related-posts/jetpack-related-posts.php]: Internal discussion: pafL3P-jr-p2

**Review by jeherve — DISMISSED** [review]
This seems to work well for me in my tests. It should be good to merge.

Just a quick, non-blocking note: in the editor as well as on the frontend, should we crop images so they always fit on a grid, just like we do for the regular related posts? This would avoid posts being unaligned like so:
image

**Comment by jeherve** [comment]
This will need a rebase first, though, to incorporate the recent changes to block registration (notably #11310).

**Comment by aldavigdis** [comment]
@jeherve It seems like videos are an exception here. But it may be a good idea to create a separate issue ticket to solve that.

One other issue that I see is that images that have not been processed correctly on wordpress.com due to the connection being bad (example: https://i0.wp.com/alda8c.pagekite.me/wp-content/uploads/2019/02/vector__527___fluttershy__28_by_dashiesparkle-daashsq.png?resize=350%2C200&ssl=1) are also squares.

I'm going to rebase this one to master and merge if all goes well.

**Review by jeherve — APPROVED** [review]
Approving once again, post rebase.

File: modules/related-posts/jetpack-related-posts.php [hunk 2]
```diff
@@ -769,7 +769,7 @@ public function get_for_post_id( $post_id, array $args ) {
 			$options['size'] = $args['size'];
 		}
 
-		if ( ! $options['enabled'] || 0 == (int)$post_id || empty( $options['size'] ) || get_post_status( $post_id) !== 'publish' ) {
```
Comment (kraftbj) [inline modules/related-posts/jetpack-related-posts.php]: @aldavigdis Could you tell me more about removing the `enabled` check? This is causing a regression in that we have a filter so folks can selectively disable RPs.

The published check was added in https://github.com/Automattic/jetpack/pull/9495 as a way to fix #9494 for a 400 HTTP request in the Block Editor for a published post.

See https://github.com/Automattic/jetpack/pull/11560

See #11546

**Pull request merged**